### PR TITLE
Sprint6 task3 categories

### PR DESCRIPTION
**Pull request by iAcay**
Hey! The code which is below was implemented in order to meet the acceptance criterias of the third task in sprint no. 6 :)

AC in this task:
-- There's a model named `Category`. A category has a title and timestamps
-- Every reward belongs to a category
-- Every category can have many rewards
-- Admin can create, edit or update any category
-- Admin can delete a category only if it has no rewards
-- Admin can add a category to any reward
-- Admin can remove a category from any reward
-- There's a system spec for admin creating a category and adding it to a reward

I have implemented one-to-many-association between categories and rewards because one of the acceptance criterias which are above says that reward belongs to a category (-> one category). 🔢  

https://user-images.githubusercontent.com/96024046/181640126-e5b8076d-3622-40e6-a219-e9f85badb77b.mov

### CODE REVIEW

File: app/views/admin_users/rewards/show.html.erb [hunk 1]
```diff
@@ -1,8 +1,16 @@
 <div class="shadow p-3 mb-5 bg-body rounded">
   <figure class="text-end">
     <figcaption class="block-footer">
-      <strong>Price:</strong>
-      <%= reward.price %>
+      <div class="row">
+      <div class="col-6 text-start">
+        <strong>Category:</strong>
+        <%= reward.category.present? ? reward.category.title : 'without category' %>
```
Comment (piowit) [inline app/views/admin_users/rewards/show.html.erb]: We are using this check `reward.category.present?` in views a few times. I would suggest moving this to the reward model method that returns `category.title` or `'without category'`

File: spec/factories/category.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :category do
+    title { 'Category Title' }
```
Comment (piowit) [inline spec/factories/category.rb]: My sugggestion is to use here sequence or `faker gem`, therefore in the tests categories will be named differently :)

**Review by piowit — COMMENTED** [review]
Nice one 👍🏻 
You can find some suggestions below 👇🏻

File: app/views/admin_users/rewards/show.html.erb (hunk 1 above)
Comment (iAcay) [reply app/views/admin_users/rewards/show.html.erb]: Thanks a lot! I changed it 👍

File: spec/factories/category.rb (hunk 2 above)
Comment (iAcay) [reply spec/factories/category.rb]: Done 👍 Now I use faker gem for creating a title :)

**Review by ciejj — APPROVED** [review]
Loos goood - no suggestions this time ;)

**Comment by iAcay** [comment]
> Loos goood - no suggestions this time ;)

Thanks!

**Pull request merged**